components/results_table/commons/results_table.lua

```diff
@@ -302,7 +305,132 @@ function BaseResultsTable:buildPlayersOnTeamOpponentConditions(opponentTeamTepla
+		:addClass('table-responsive')
+		:node(displayTable)
+end
+
+function BaseResultsTable:_buildRows(placementData)
+	local rows = {}
+
+	if placementData.header then
+		table.insert(rows, mw.html.create('tr')
+			:tag('th'):addClass('sortbottom'):attr('colspan', 42):wikitext(placementData.header):done()
+			:done())
+	end
+
+	for _, placement in ipairs(placementData) do
+		table.insert(rows, self:buildRow(placement))
+	end
+
+	return rows
+end
+
+-- overwritable
+function BaseResultsTable:rowHighlight(placement)
+	if String.isNotEmpty(placement.publishertier) then
+		return 'valvepremier-highlighted'
+	end
+end
+
+-- overwritable
+function BaseResultsTable:tierDisplay(placement)
+	local tierDisplay
+
+	if String.isEmpty(placement.liquipediatiertype) and String.isEmpty(placement.liquipediatier) then
+		return '', ''
+	elseif String.isNotEmpty(placement.liquipediatiertype) then
+		local tierType = placement.liquipediatiertype:lower()
+		tierDisplay = Tier.text.types[tierType] or placement.liquipediatiertype
+	else
+		tierDisplay = Tier.text.tiers[placement.liquipediatier] or placement.liquipediatier
+	end
+
+	return Page.makeInternalLink(
+		{},
+		tierDisplay,
+		tierDisplay .. ' Tournaments'
+	), tierDisplay
+end
+
+-- overwritable
+function BaseResultsTable:opponentDisplay(data, options)
+	options = options or {}
+
+	if not data.opponenttype then
+		return OpponentDisplay.BlockOpponent{
+			opponent = Opponent.tbd(),
+			flip = (options or {}).flip
+		}
+	elseif (data.opponenttype ~= Opponent.solo or not options.teamForSolo)
+		and data.opponenttype ~= Opponent.team then
+
+		return OpponentDisplay.BlockOpponent{
+			opponent = Opponent.fromLpdbStruct(data),
+			flip = (options or {}).flip
+		}
+	end
+
+	local teamTemplate
+	if data.opponenttype == Opponent.team then
+		teamTemplate = data.opponenttemplate
+	else
+		teamTemplate = data.opponentplayers.p1team
+	end
+
+	if String.isEmpty(teamTemplate) then
+		return
+	end
+
+	return Team.icon(nil, teamTemplate, data.date)
+end
+
+-- overwritable
+-- shadows the current implementation
+-- todo: in the future change it to use darkmode icons too
+-- needs upgrading the game icon data modules first though
+function BaseResultsTable:gameIcon(placement)
+	local gameIcon = self.config.gameIconsData[placement.game] or 'Logo filler event.png'
+	gameIcon = gameIcon:gsub('File:', '')
+	return LeagueIcon.display{
+		icon = gameIcon,
+		options = {noTemplate = true, noLink = true},
+	}
+end
+
+function BaseResultsTable.tournamentDisplayName(placement)
+	if String.isNotEmpty(placement.tournament) then
+		return placement.tournament
+	end
+
+	return placement.pagename:gsub('_', ' ')
+end
+
+function BaseResultsTable:buildHeader()
+	error('Function "buildHeader" needs to be set via the module that requires "Module:BaseResultsTable/Base"')
+end
+
+function BaseResultsTable:buildRow(placement)
+	error('Function "buildRow" needs to be set via the module that requires "Module:BaseResultsTable/Base"')
+end
```

Comment:
Don't want to provide basic versions of these? Most wikis would be happy with that I believe

Reply:
i want to use this module for both results/achievements and awards
for awards the display is different so the specific modules set those 2 functions
not intended to move them to wiki specific customs